pkg/fileservice/counter.go

```diff
@@ -45,9 +50,12 @@ func updateCounters(ctx context.Context, fn func(*Counter)) {
 	}
 }
 
-func WithCounter(ctx context.Context, counter *Counter) context.Context {
+func WithCounter(ctx context.Context, statsFName string, counter *Counter) context.Context {
 	// check existed
 	v := ctx.Value(CtxKeyCounters)
+
+	metric.DefaultStatsRegistry.RegisterStats(statsFName, counter) //TODO: Need suggestions here.
```

Comment:
I think the `metric` package should implement its own WithCounter and call this WithCounter to set things up.

Reply:
WithCounter may be called in every SQL statement or even every CN-DN request, the `metric` package should not register all of them.

Reply:
Got it.